### PR TITLE
allow reading body from request and reply to con with type = ack

### PR DESCRIPTION
**Pull request by johngiffin**

### CODE REVIEW

File: lib/coap/phoenix/conn.ex [hunk 1]
```diff
@@ -55,6 +56,10 @@ defmodule CoAP.Phoenix.Conn do
     {:ok, nil, req}
   end
 
+  def read_req_body(state, opts) do
```
Comment (johngiffin) [inline lib/coap/phoenix/conn.ex:59]: This gets called here by `read_body`: https://github.com/elixir-plug/plug/blob/master/lib/plug/conn.ex#L1034

File: lib/coap/message.ex [hunk 2]
```diff
@@ -204,7 +204,7 @@ defmodule CoAP.Message do
 
   def response_for(%__MODULE__{type: :con} = message) do
     %__MODULE__{
-      type: :con,
```
Comment (tpitale) [inline lib/coap/message.ex:207]: This con never seemed right to me, but I'm pretty sure I copied it from `gen_coap`. Happy to merge this and find out! 😄

File: lib/coap/phoenix/conn.ex [hunk 3]
```diff
@@ -12,7 +12,8 @@ defmodule CoAP.Phoenix.Conn do
       headers: headers,
       qs: qs,
       peer: {remote_ip, _},
-      owner: owner
+      owner: owner,
+      message: message
```
Comment (tpitale) [inline lib/coap/phoenix/conn.ex:16]: Should we store just the `message.payload` as body.

**Review by tpitale — APPROVED** [review]

File: lib/coap/message.ex (hunk 2 above)
Comment (johngiffin) [reply lib/coap/message.ex:207]: Hmm, strange.  gen_coap responds with ack as far as I can tell

File: lib/coap/phoenix/conn.ex (hunk 3 above)
Comment (johngiffin) [reply lib/coap/phoenix/conn.ex:16]: Sure, I started out that way but then saw message was already available.  Happy to change it.

**Pull request merged**